add support for ProxySQL v2.x Galera support

unki opened the pull request:
This PR adds support for the upcoming ProxySQL v2.x Galera support, where a new table `mysql_galera_hostgroups` gets introduced.

<https://github.com/sysown/proxysql/wiki/Main-(runtime)#mysql_galera_hostgroups>

Basically I just copied everything _proxysql's_ `proxy_mysql_replication_hostgroup` Puppet type and reworked it for `mysql_galera_hostgroups` as `proxy_mysql_galera_hostgroup`.

bastelfreak commented:
Hi @unki, thanks for the PR! Can you take a look at the failing travis jobs? Please also have a look at the used email address in your commit, it isn't associated with your github account.

unki commented:
I've updated the PR and added additional setters. I'm not sure if `mk_resource_methods` should actually take care of this, but otherwise I do not get the galera-settings been updated in ProxySQL.

alexjfisher commented:
Superseded by https://github.com/voxpupuli/puppet-proxysql/pull/103
